Reorganize the "Home" section of the docs

ptgott opened the pull request:
Fixes #13206

The organization of the "Home" section of the docs currently says little
about its purpose, making it more difficult for new visitors to the docs
to find content. I've proposed changing this to a "Get Started" section
that includes content related to trying out Teleport and choosing an
edition.

(Re)move pages that are unrelated to getting started:

- Rename the "Preview" section to "Releases" and move the Changelog
  there. The Preview section doesn't include preview content very often,
  and usually only includes the Upcoming Releases page. I think it makes
  sense to devote this section to releases.

- Remove the Adopters page and redirect to the Teleport case studies
  page. The Adopters page receives close to no traffic and is
  infrequently updated, so keeping it risks giving users outdated
  information.

- Remove the Production guide. Redirects currently make it impossible
  to navigate to this page.

- Move the Digital Ocean guide to "/docs/setup/deployments", since other
  platform-specific deployment guides are also placed there.

Organize content related to getting started with Teleport into "Try out
Teleport" and "Deploy a Teleport Cluster" subsections:

- Encourage new users to try a quick lab by including a new menu page
  with links to all of our Instruqt tracks and local demo environments.
  Move our demo environment guides into
  "getting-started/try-out-teleport"

- Put all of our edition-specific Getting Started guides into one
  subsection, "getting-started/deploy-a-cluster". This means that
  the Linux Server guide is no longer the de facto entryway into
  Teleport, which can confuse Cloud and Enterprise users.

  The "Deploy a Cluster" subsection includes a menu page with a guide
  and  comparison table for Teleport's editions.

- With these new subsections, there's not really a place anymore for the
  /getting-started.mdx page, so I have removed this.

ptgott committed:
Reorganize the "Home" section of the docs

Fixes #13206

The organization of the "Home" section of the docs currently says little
about its purpose, making it more difficult for new visitors to the docs
to find content. I've proposed changing this to a "Get Started" section
that includes content related to trying out Teleport and choosing an
edition.

(Re)move pages that are unrelated to getting started:

- Rename the "Preview" section to "Releases" and move the Changelog
  there. The Preview section doesn't include preview content very often,
  and usually only includes the Upcoming Releases page. I think it makes
  sense to devote this section to releases.

- Remove the Adopters page and redirect to the Teleport case studies
  page. The Adopters page receives close to no traffic and is
  infrequently updated, so keeping it risks giving users outdated
  information.

- Remove the Production guide. Redirects currently make it impossible
  to navigate to this page.

- Move the Digital Ocean guide to "/docs/setup/deployments", since other
  platform-specific deployment guides are also placed there.

Organize content related to getting started with Teleport into "Try out
Teleport" and "Deploy a Teleport Cluster" subsections:

- Encourage new users to try a quick lab by including a new menu page
  with links to all of our Instruqt tracks and local demo environments.
  Move our demo environment guides into
  "getting-started/try-out-teleport"

- Put all of our edition-specific Getting Started guides into one
  subsection, "getting-started/deploy-a-cluster". This means that
  the Linux Server guide is no longer the de facto entryway into
  Teleport, which can confuse Cloud and Enterprise users.

  The "Deploy a Cluster" subsection includes a menu page with a guide
  to different editions.

- With these new subsections, there's not really a place anymore for the
  /getting-started.mdx page, so I have removed this.

zmb3 commented:
w.r.t 

> Rename the "Preview" section to "Releases" and move the Changelog
there. The Preview section doesn't include preview content very often,
and usually only includes the Upcoming Releases page. I think it makes
sense to devote this section to releases.

I think we're actually planning to use the preview session more and provide more clear messaging about features that are in preview. Maybe it makes sense to keep this section.
@klizhentas what do you think?

ptgott committed:
Respond to PR feedback

Restore the "Preview" section and Changelog page, since the
Preview section will be used for more preview content after all.

ptgott committed:
Revert changelog links to be relative to the docs

ptgott commented:
Converted to a draft since we will be revisiting the reorganization plan for the docs. Keepings this open in case we can modify it to accommodate the new plan.

ptgott commented:
Closing in favor of https://github.com/gravitational/teleport/pull/15357
